server/vertx/src/main/scala/sttp/tapir/server/vertx/encoders/package.scala

```diff
@@ -15,8 +15,10 @@ package object encoders {
     */
   private[vertx] def inputStreamToBuffer(is: InputStream): Buffer = {
     val buffer = Buffer.buffer()
+    val buf = new Array[Byte](bufferSize)
     while (is.available() > 0) {
-      buffer.appendBytes(is.readNBytes(bufferSize))
+      val read = is.read(buf)
+      buffer.appendBytes(buf, 0, read)
```

Comment:
Are you sure it's gonna work?
Always appending to the start of the buffer ?

Reply:
oh ... I though this specifies the slice of `buf` that will be appended

Reply:
and as the tests passed ... ;)

Reply:
yes, this seems correct, see the javadocs to the delegating impl (`io.netty.buffer.ByteBuf.writeBytes`) and e.g. the implementaiton in `AbstractByteBuf`:

```
@Override
    public ByteBuf writeBytes(byte[] src) {
        writeBytes(src, 0, src.length);
        return this;
    }
```